### DISCUSSION TITLE
Question: Override default postinstall script - for custom path

### DISCUSSION THREAD

**Discussion by harley-codes** [prisma/prisma#17897] [Q&A] [answered]
I am using NextJS to build my application. When running a build, prisma triggers a postinstall script.

However, I use a custom path so this generates a warning. 
The simplest solution was to add my own postinstall script to package.json that call 'prisam generate --schema=...' 

My question is if there is a more elegent way of handling this? As prisma generate is being run twice, and will always output a warning in my logs for the default. 

I love Prisma, it's a great product. Keep on rocking! 😎

**Comment by nurul3101** [comment, marked as answer]
Hello @Harley-Torrisi 👋 

Thank you for your kind words, and we are glad to hear that you love Prisma!

Can you please elaborate on what warning do you get when invoking the build command?

**Reply by harley-codes** [reply]
Hmm... Something very intersting has happened!

At the time of posting this question, this was my last build log output regarding prisma.

- > hobby-cms@0.1.0 postinstall
- 00:12:39.471 | > prisma generate \|\| true && npm run db:prisma-cockroach:generate
- 00:12:40.407 | prisma:warn The postinstall script automatically ran `prisma generate` and did not find your `prisma/schema.prisma`.
- 00:12:40.407 | If you have a Prisma schema file in a custom path, you will need to run
- 00:12:40.408 | `prisma generate --schema=./path/to/your/schema.prisma` to generate Prisma Client.
- 00:12:40.408 | If you do not have a Prisma schema file yet, you can ignore this message.
- 00:12:40.875 | > hobby-cms@0.1.0 db:prisma-cockroach:generate
- 00:12:40.876 | > prisma generate --schema=src/scripts/modules/database/vendors/prisma-cockroach/schema.prisma
- 00:12:41.769 | Prisma schema loaded from src/scripts/modules/database/vendors/prisma-cockroach/schema.prisma

I just ran a new build and it has changed.
- > hobby-cms@0.1.0 postinstall
- 02:39:25.747 | > npm run db:prisma-cockroach:generate
- 02:39:26.049 | > hobby-cms@0.1.0 db:prisma-cockroach:generate
- 02:39:26.049 | > prisma generate --schema=src/scripts/modules/database/vendors/prisma-cockroach/schema.prisma
- 02:39:26.555 | Prisma schema loaded from src/scripts/modules/database/vendors/prisma-cockroach/schema.prisma

Im using Vercel for deployment. From a package.json standpoint, nothing was updated between those two builds.
Note sure what changed; either vercel has some kind of leanring in place, or Prisma detected I added a post install 🤷‍♂️

Either way, its a win 😎

**Reply by nurul3101** [reply]
I am glad to hear that the issue is resolved 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
